paasta_tools/api/client.py

```diff
@@ -89,16 +91,41 @@ def __init__(
         system_paasta_config: SystemPaastaConfig,
     ) -> None:
         if scheme == 'https':
+            ca_cert_path = None
             if system_paasta_config.get_enable_client_cert_auth():
+                ecosystem = system_paasta_config.get_vault_cluster_config()[cluster]
```

Comment:
This seems to be assume that we use the same cert/key/ca for all paasta clusters in the ecosystem, right?

Reply:
Yeah, I think this is ok to start with. I was going back and forth between cluster/ecosystem. But as this is just authn I think ecosystem will be fine and per-cluster permission can be at the authz layer.